src/_pytest/main.py

```diff
@@ -805,21 +809,6 @@ def collect(self) -> Iterator[Union[nodes.Item, nodes.Collector]]:
                     self._notfound.append((report_arg, col))
                     continue
 
-                # If __init__.py was the only file requested, then the matched
```

Comment:
This entire thing was only needed because of the `__init__.py`, no longer relevant